controllers/medusa/medusabackupjob_controller.go

```diff
@@ -210,7 +215,25 @@ func (r *MedusaBackupJobReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return ctrl.Result{RequeueAfter: r.DefaultDelay}, nil
 }
 
-func (r *MedusaBackupJobReconciler) createMedusaBackup(ctx context.Context, backup *medusav1alpha1.MedusaBackupJob, logger logr.Logger) error {
+func (r *MedusaBackupJobReconciler) getBackupSummary(ctx context.Context, backup *medusav1alpha1.MedusaBackupJob, pods []corev1.Pod, logger logr.Logger) (*medusa.BackupSummary, error) {
+	for _, pod := range pods {
+		if remoteBackups, err := GetBackups(ctx, &pod, r.ClientFactory); err != nil {
```

Comment:
I think it's because the BackupSummary contains information from all nodes, while BackupStatus is for an individual node. And yeah, it's definitely overkill to read all backups just to get one (it can be quite expensive when you have a lot of backups in storage).
It would be good to be able to filter server side and get a single backup info though. The `medusa status` command already implements reading a [single backup](https://github.com/thelastpickle/cassandra-medusa/blob/master/medusa/status.py#L30) to get completion information. We could enrich it with the topology info and expose it in the GRPC server for use here.
I'll create a refactoring ticket for this.